android/src/main/java/com/swmansion/reanimated/layoutReanimation/ScreensHelper.java

```diff
@@ -0,0 +1,66 @@
+package com.swmansion.reanimated.layoutReanimation;
+
+import android.view.View;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ScreensHelper {
+
+  public static View getTabNavigator(View view) {
+    View currentView = view;
+    while (currentView != null) {
+      if (currentView.getClass().getSimpleName().equals("ScreenContainer")) {
+        return currentView;
+      }
+      if (currentView.getClass().getSimpleName().equals("Screen")
+          && currentView.getParent() != null
+          && currentView
+              .getParent()
+              .getClass()
+              .getSimpleName()
+              .equals("ScreensCoordinatorLayout")) {
+        View screen = currentView;
+        Class<?> screenClass = screen.getClass();
+        try {
+          Method getContainer = screenClass.getMethod("getContainer");
+          currentView = (View) getContainer.invoke(screen);
+        } catch (NoSuchMethodException
+            | InvocationTargetException
+            | IllegalAccessException ignored) {
+        }
```

Comment:
What happens if this exception is thrown? I think it could lead to infinite loop. Maybe let's store the previous `currentView` and break the loop if it doesn't change during the iteration?